### ISSUE TITLE
Build R package from source error: no packages specified

### ISSUE THREAD

**Issue by JiaRu2016** [dmlc/xgboost#5565]
I want to build and install xgboost R package from source, following steps 
in https://github.com/dmlc/xgboost/blob/master/doc/build.rst#r-package-installation

```
git clone --recursive https://github.com/dmlc/xgboost
cd xgboost
git submodule init
git submodule update
mkdir build
cd build
cmake .. -DR_LIB=ON
make -j$(nproc)
make install
```

the last  step `make install` throws Error:

![Screen Shot 2020-04-20 at 10 00 35](https://user-images.githubusercontent.com/17561508/79707055-cc9e2e00-82ed-11ea-8355-524f7b1c1613.png)

I'v tried to build shared library seperately and it works well, so it looks like the problem it specific to R package.

Environment Info:
System: Ubuntu 18.04
R version:  3.6.2
xgboost branch: release_1.0.0

**Comment by trivialfis** [comment]
I can't reproduce ..

**Comment by JiaRu2016** [comment]
I'v tried: 

```
cd path/to/xgboost/R-package
R CMD INSTALL .
```

it seems installed succesfully. but I wonder Is it another "correct"  way to install R package?

**Comment by trivialfis** [comment]
This is the CRAN way of installing R package, which means you will get a strictly built R package with minimal functionality.

**Comment by hcho3** [comment]
"minimal functionality" is not quite the right word. It just means that you won't have access to `gpu_hist` and other features that involves the use of NVIDIA GPUs.

**Comment by trivialfis** [comment]
@hcho3 Nor cpu prefetching I assume?

**Comment by hcho3** [comment]
@trivialfis Yes.

**Comment by hcho3** [comment]
Fixed by #5764
